### PR TITLE
Adhere to runner guidelines

### LINKED ISSUE #28: Adhere to runner guidelines
Use async API and other things:

https://github.com/postcss/postcss/blob/master/Runner%20Guidelines.md

### PR DESCRIPTION
**Pull request by kevva**
Fixes #28.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ it('should generate source maps', function (cb) {
 		.pipe(write);
 
 	write.on('data', function (file) {
-		assert.equal(file.sourceMap.mappings, 'AAAA;CACC,eAAc;EACd');
+		assert.equal(file.sourceMap.mappings, 'AAAA;CACC,cAAc;CACd');
```
Comment (kevva) [inline test.js]: Dunno why this needed to be changed.

File: index.js [hunk 2]
```diff
@@ -19,27 +20,33 @@ module.exports = function (opts) {
...
 		var fileOpts = objectAssign({}, opts);
-
-		try {
-			res = autoprefixer(fileOpts).process(file.contents.toString(), {
+		var processor = postcss()
+			.use(autoprefixer(fileOpts))
+			.process(file.contents.toString(), {
 				map: file.sourceMap ? {annotation: false} : false,
 				from: file.relative,
 				to: file.relative
```
Comment (sindresorhus) [inline index.js]: looks like these should be the full path now https://github.com/postcss/postcss/blob/master/Runner%20Guidelines.md#21-set-from-and-to-processing-options

**Comment by sindresorhus** [comment]
Missing this https://github.com/postcss/postcss/blob/master/Runner%20Guidelines.md#31-dont-show-js-stack-for-csssyntaxerror and https://github.com/postcss/postcss/blob/master/Runner%20Guidelines.md#43-postcssrunner-keyword-in-packagejson

**Comment by kevva** [comment]
Do you want to show the stack at all? ` gutil.PluginError` hides it by default. Will add the keyword.

**Comment by sindresorhus** [comment]
> gutil.PluginError hides it by default.

Nope, just checked. It's shown by default.

Just deal with it the same as  https://github.com/postcss/gulp-postcss/blob/master/index.js#L6

**Comment by kevva** [comment]
Weird, it says that it's not shown by default in their [documentation](https://github.com/gulpjs/gulp-util#new-pluginerrorpluginname-message-options).

> By default the stack will not be shown. Set options.showStack to true if you think the stack is important for your error.

**Comment by sindresorhus** [comment]
@kevva But try it in practice. It shows the stack by default.

**Comment by kevva** [comment]
@sindresorhus, do you want to show the source code too (like https://github.com/postcss/gulp-postcss/blob/master/index.js#L80)?

**Comment by kevva** [comment]
@sindresorhus, alright, should be good now.

**Comment by sindresorhus** [comment]
Awesomesauce :)

**Pull request merged**